Source/Provers/SMTLib/SMTLibInteractiveTheoremProver.cs

```diff
@@ -212,7 +202,7 @@ public override async Task<Outcome> CheckOutcome(ErrorHandler handler, int error
           {
             errorsDiscovered++;
 
-            result = await GetResponse(cancellationToken);
+            result = await CheckSatAndGetResponse(cancellationToken);
```

Comment:
I'm worried that this might have slightly different semantics than before. The original coded seems to be assuming that a request was sent before entering this method, and that it can first get a response to that request before issuing a further `(check-sat)`, though the logic here is complex and I'm not entirely sure of that.

Reply:
First it was:

```
send check-sat
// enter method
while(true) {
  receive response
  if (go another round)
    send check-sat
  else
    break
}
```

Now it's

```
// enter method
while(true) {
  send check-sat
  receive response
  if (not go another round)
    break
}
```

Reply:
Ah, okay. I think the first check-sat was far enough away I wasn't sure it was there. Thanks!